hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/spi/JdbcTypeRegistry.java

```diff
@@ -174,6 +178,62 @@ public AggregateJdbcType findAggregateDescriptor(String typeName) {
 		return aggregateDescriptorMap.get( typeName.toLowerCase( Locale.ROOT ) );
 	}
 
+	public JdbcType resolveTypeConstructorDescriptor(
```

Comment:
Can we please add some Javadoc here.  Why the map of maps?

Reply:
You're right, it's better to have a single key object. I fixed that and added some java doc.